### ISSUE TITLE
Error with fallback signature and/or README documentation

### ISSUE THREAD

**Issue by profburke** [IBM-Swift/CircuitBreaker#37]
This seems to be a known issue and one that, in theory, has already been addressed (see #16). But I'm still having issues with the signature for, and invocation of, the fallback function.

The README indicates that the fallback functions's signature should match

    (<BreakerError, (fallbackArg1, fallbackArg2,...)>) -> Void

(*Unless I'm missing something, the angle brackets are a typo in the README.*)

and that the `run` method on `CircuitBreaker` should be called as follows:

    breaker.run(commandArgs: (a: 10, b: 20), fallbackArgs: (msg: "Something went wrong."))

However, if I create a fallback with the signature `(BreakerError, String) -> Void` and try to invoke `run` as indicated above, I get the following error message:

    error: cannot convert value of type '(msg: String)' to expected argument type 'String' 
    breaker.run(commandArgs: (a: 12, b: 13), fallbackArgs: (msg: "This is a test"))                                                                                
                                                       ^~~~~~~~~~~~~~~~~~~~~~~                                                                                 
                                                                               as! String  

Further, if I create a fallback with the signature `(BreakerError, String, String) -> Void` and try to invoke `run` with `breaker.run(commandArgs: (a: 10, b: 20), fallbackArgs: (msg, "First Message", msg2: "SecondMessage:))` I receive the following error:

    error: cannot convert value of type '(Int, Int) -> Int' to expected argument type '(_) -> (_)'
    let breaker = CircuitBreaker(command: myFunc, fallback: myFallback)                                                                                            
                                      ^~~~~~                                                

(*Note: the error is with parsing the **command**, not the fallback.*)

I can, however, successfully invoke `run` as follows:

     breaker.run(commandArgs: (a: 10, b: 20), fallbackArgs: "Something went wrong.")

I haven't yet finished working my way through the source, so that might clear up some mis-understanding on my part. Nevertheless, it seems that the README is definitely incorrect.
Also, it would be very useful to actually be able to pass more than one argument to the fallback function.

**Comment by rolivieri** [comment]
@profburke Thanks for reaching out. Yes, we did have a few documentation issues in the README. We just made a few changes to address those.

Let’s say you have the following command and fallback functions:

```
func sum(a: Int, b: Int, c: String) -> Void {
    print(a + b)
    sleep(40)
}

func fallbackFunction(error: BreakerError, str: String) -> Void {
    print("str: \(str)")
}
```

Then instantiating a `CircuitBreaker` instance and running it would look like this:

```
let breaker1 = CircuitBreaker(command: sum, fallback: fallbackFunction)
let str1: String = "str1"
breaker1.run(commandArgs: (30, 49, str1), fallbackArgs: str1)
```

If you have the following command and fallback functions (i.e. need to pass multiple parameters passed to the fallback function after the BreakerError parameter):

```
func sum(a: Int, b: Int, c: String) -> Void {
    print(a + b)
    sleep(40)
}

func anotherFallback(error: BreakerError, strs: (str1: String, str2: String)) -> Void {
    print("str1: \(strs.str1), str2: \(strs.str2)")
}
```

Then instantiating a `CircuitBreaker` instance and running it would look like this:

```
let breaker2 = CircuitBreaker(command: sum, fallback: anotherFallback)
let str2: String = "str2"
breaker2.run(commandArgs: (a: 40, b: 100, str1), fallbackArgs: (str1: str1, str2: str2))

```

In other words, if you need your fallback function to receive more than one value for the second parameter (`C`), then you need to define a tuple and pass the values inside the tuple (as shown above.

Hope this helps. Let us know if you have any more questions.

**Comment by profburke** [comment]
Thanks for the response. I did puzzle out sending a tuple as the second parameter to my fallback function, but it's good to know that's the "proper" solution.

I do have some questions regarding using the library. Is there a forum or mail list somewhere? Or should I just open more issues on the repo?

**Comment by rolivieri** [comment]
@profburke You can reach us on slack @ `https://swift-at-ibm-slack.mybluemix.net/`. Feel free to contact me directly over there or use the general channel. You can use GH issues as well.

**Comment by profburke** [comment]
Thanks. I'll reach out on Slack.
